src/LinearSolverTypes.f90

```diff
@@ -1348,6 +1371,105 @@ SUBROUTINE setConv_LinearSolverType_Iterative(solver,normType_in,  &
     ENDSUBROUTINE setConv_LinearSolverType_Iterative
 !
 !-------------------------------------------------------------------------------
+!> @brief Sets the convergence criteria for PETSc iterative solvers
+!> @param solver The linear solver to act on
+!> @param normType An integer representing the convergence check norm
+!> @param rTol Relative convergence criteria
+!> @param absTol Absolute convergence criteria
+!> @param dTol Divergence criteria
+!> @param maxIters The maximum number of iterations to perform
+!> @param nRestart The number of iterations before GMRES restarts
+!>
+!> This subroutine sets the convergence criterion for the iterative solver.
+!>
+    SUBROUTINE setConvPETSc(solver,normType_in,rTol_in,maxIters_in,nRestart_in, &
+                                   absTol_in,dTol_in)
```

Comment:
I could be missing something, but it looks like this interface is basically identical to setConv_LinearSolverType_Iterative and doing a lot of similar things.  Do they really need to be 2 separate methods on the class?  If they do, I'd prefer not having a method that includes a specific TPL name (i.e., replace the PETSC part of setConvPETSC with something that might help help describe why this routine is different from the one immediately above).

My preference would be to just fold this functionality into the old setConv method if you can.

Reply:
That makes sense; I'll merge them into the same subroutine.  When I made this code, it was before Shane's recent commit that changed the standard setConv so that the rel and abs convergence tolerances were split.  Now they're practically the same.